projects/plugins/jetpack/class.jetpack.php

```diff
@@ -5277,8 +5280,10 @@ public function post_login_form_to_signed_url( $url, $path, $scheme ) {
 	 * Make sure the POSTed request is handled by the same action.
 	 */
 	public function preserve_action_in_login_form_for_json_api_authorization() {
+		$http_host   = isset( $_SERVER['HTTP_HOST'] ) ? sanitize_text_field( wp_unslash( $_SERVER['HTTP_HOST'] ) ) : '';
+		$request_uri = isset( $_SERVER['REQUEST_URI'] ) ? sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ) ) : '';
```

Comment:
sanitize_text_field() will probably break things here, as it strips out any percent-encoded characters.

Really for both of these lines the `esc_url()` below is the needed sanitization.

Reply:
2dc7e0d8c1a9999981b1cd3b2d05279a6b94fdbb